File: V2/hono_backend/api_backend/src/redis.ts
import { env } from "process";
import { createClient } from "redis";

let client: any;
let subscriber: any;
async function redisClient() {
  const redis_url = env.REDIS_URL || "redis://localhost:6379"

  client = createClient({ url: redis_url })
  subscriber = createClient({ url: redis_url })

  await client.connect()
  await subscriber.connect()

  client.on('error', (err: Error) => console.log("Redis Client Error", err))
}

redisClient()

export { client, subscriber }

File: V3/worker_engine/internal/redis/redis.go
package redis

import (
	"context"
	"fmt"
	"os"

	"github.com/redis/go-redis/v9"
)

type Client struct {
	*redis.Client
}

func NewClient() (*Client, error) {
	redisAddr := os.Getenv("REDIS_ADDR")
	if redisAddr == "" {
		redisAddr = "localhost:6379"
	}

	rdb := redis.NewClient(&redis.Options{
		Addr: redisAddr,
	})

	ctx := context.Background()
	_, err := rdb.Ping(ctx).Result()
	if err != nil {
		return nil, fmt.Errorf("failed to connect to Redis: %w", err)
	}

	return &Client{rdb}, nil
}
